### PR TITLE
Add events and ec track

### PR DESCRIPTION
**Pull request by tmieslinger**
This PR adds next to the segments also a list of `events` and loads and plots the EC track in the template.

### CODE REVIEW

**Comment by tmieslinger** [comment]
I added the attribute `distance` that can be added to an `event`. I would assume that we usually want to note down the distance to the target for overpasses and underpasses and it's nice to have it as an attribute to be able to extract the info easily.

**Comment by tmieslinger** [comment]
@Lucalino The last chunk of changes mostly add two features:
* circle center coordinates and radius are added automatically as soon as a kinds=`circle` is defined
* the EC meeting point is added to the `events` list. This closes issue #54 and  #68.
* some clean up

I hope we are at a state now that we have most relevant infos at hand and can start segmenting all flights ;-)

**Review by Lucalino — CHANGES_REQUESTED** [review]
Overall, this request adds super useful new features and most things work as expected. The only major trouble I had was with the `overpass_point` function which seems to require more input than defined (`seg`). But perhaps I overlooked/didn't understand something in which case an update of the accompanying information would be great. 

Apart from this issue, I recommend adding a few more cells were typical overpass calculations are performed so that not everyone needs to piece these code snippets together by themselves.

File: scripts/segmentation_template.md [hunk 1]
```diff
@@ -174,48 +186,70 @@ gs = fig.add_gridspec(2,2)
...
+plt.ylabel("latitude / °")
+plt.legend();
+```
+
+## Events
+events are different from segments in having only **one** timestamp. Examples are the usual "EC meeting points" or station / ship overpasses. In general, events include a mandatory `event_id` and `time`, as well as optional statements on `name`, a list of `kinds`, the `distance` in meters, and a list of `remarks`. Possible `kinds`include:
+- `ec_underpass`
+- `meteor_overpass`
+- `bco_overpass`
+- `cvao_overpass`
```
[the remark this answers is not in the document]
Comment (tmieslinger) [reply scripts/segmentation_template.md:236]: yes, the [CVAO](https://orcestra-campaign.org/cvao.html) is the ground station near Mindelo. As we also attribute the BCO by it's actual name I thought that it's consistent to name the other station accordingly.

File: scripts/utils.py [hunk 2]
```diff
@@ -23,27 +28,168 @@ def get_sondes_l1(flight_id):
                 for filename in filenames]
     return np.array([d["launch_time"].values for d in datasets])
 
-def get_overpass_info(seg, ds, target_lat, target_lon):
+def get_overpass_point(ds, target_lat, target_lon):
```
[the remark this answers is not in the document]
Comment (tmieslinger) [reply scripts/utils.py:31]: that was a bug :) thanks for pointing out. It's corrected now.

File: scripts/segmentation_template.md [hunk 3]
```diff
@@ -174,48 +186,70 @@ gs = fig.add_gridspec(2,2)
...
+
+The `event_id` will be added when saving it to YAML.
+
+The EC underpass event can be added to a list of events via the function `ec_event`.
+
+```python
+events = [
+    ec_event(ds, ec_track),
```
[the remark this answers is not in the document]
Comment (tmieslinger) [reply scripts/segmentation_template.md:244]: * I added another example event in from of a dictionary in the template file
* BCO and CVAO can easily be loaded via `from orcestra import bco, mindelo`. We could add that by default to the cell that loads all modules. So far, I left it out as not all flights will need it.
* METEOR and PACE tracks: would be nice to add. Maybe the first person segmenting a flight that includes such segments can work on that?

File: scripts/segmentation_template.md [hunk 4]
```diff
@@ -28,67 +28,79 @@ import pandas as pd
 import matplotlib.pyplot as plt
 
 from navdata import get_navdata_HALO
-from utils import get_sondes_l1, seg2yaml, get_takeoff_landing
+from utils import *
```
[the remark this answers is not in the document]
Comment (tmieslinger) [reply scripts/segmentation_template.md:31]: see comments above. This is covered by the commits adding PACE and METEOR tracks.

File: scripts/segmentation_template.md [hunk 5]
```diff
@@ -28,67 +28,79 @@ import pandas as pd
...
+dist_ec, t_ec = get_overpass_track(ds, ec_track)
 ```
 
+## Overview plot: HALO track, EC meeting point, and dropsonde locations
+
 ```python
-ds["roll"].hvplot()
+plt.plot(ds.lon.sel(time=slice(takeoff, landing)), ds.lat.sel(time=slice(takeoff, landing)), label="HALO track")
+plt.scatter(ds_drops.lon, ds_drops.lat, s=10, c="k", label="dropsondes")
+plt.plot(ec_track.lon, ec_track.lat, c='C1', ls='dotted')
```
[the remark this answers is not in the document]
Comment (tmieslinger) [reply scripts/segmentation_template.md:93]: You found the one and only research flight that explicitly and by design did not try to meet EC, nor any other satellite and spontaneously also not the METEOR ship :D
[-> see flight report](https://orcestra-campaign.org/reports/HALO-20240821a.html) without any sign of EC

[the remark this answers is not in the document]
Comment (tmieslinger) [reply scripts/segmentation_template.md:93]: However, I did add a "refinement" to the `get_ec_track` function to select the EC track data for the time between aircraft takeoff and landing. Seems cleaner than the hack before.

[the remark this answers is not in the document]
Comment (Lucalino) [reply scripts/segmentation_template.md:93]: Haha, sorry for not checking the flight report more carefully. What a coincidence. 😅

File: scripts/segmentation_template.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tmieslinger) [reply scripts/segmentation_template.md:265]: hm... so the way the PACE track was derived during the campaign (and also the way it's currently implemented) does not work well after the actual overpass event as it's becoming more uncertain and drifts away from the actual track with time passing. This needs to be changed to load the true PACE track infos for dates in teh past!

**Comment by Lucalino** [comment]
Thank you for the updates! I think, everything is quite clear now and I'll merge the branch. Happy segmenting everyone! 🙌

**Comment by tmieslinger** [comment]
As mentioned above, the PACE track is incorrect (drifts with time). I am on it and will create a separate PR with a correction.

**Pull request merged**